### PR TITLE
Allow ci command to specifiy how many ports to install. Also general code cleanup.

### PR DESCRIPTION
**Pull request by cbezault**

### CODE REVIEW

File: ports/sdl2-gfx/CMakeLists.txt [hunk 1]
```diff
@@ -2,7 +2,12 @@ cmake_minimum_required(VERSION 3.0)
 project(SDL2_GFX C)
 
 find_path(SDL_INCLUDE_DIR SDL2/SDL.h)
-find_library(SDL_LIBRARY NAMES SDL2d SDL2)
+find_library(SDL_LIBRARY_RELEASE NAMES SDL2)
```
Comment (ras0219-msft) [inline ports/sdl2-gfx/CMakeLists.txt]: Can these changes be split into a separate PR?

Comment (cbezault) [reply ports/sdl2-gfx/CMakeLists.txt]: Yeah, didn't realize they would, get auto-magically pulled in here.
Working on splitting them out.

File: toolsrc/src/vcpkg/commands.ci.cpp [hunk 2]
```diff
@@ -391,99 +479,69 @@ namespace vcpkg::Commands::CI
...
+                        result.spec.to_string(),
+                        result.build_result.code,
+                        result.timing,
+                        result.get_binary_paragraph() ?
+                            result.get_binary_paragraph()->abi : "");
                 }
 
                 // Adding results for ports that were not built because they have known states
-                for (auto&& port : split_specs->known)
+                for (auto&& [spec, result] : split_specs.known)
```
Comment (ras0219-msft) [inline toolsrc/src/vcpkg/commands.ci.cpp]: Note: need to check if this is supported in VS2015

Comment (cbezault) [reply toolsrc/src/vcpkg/commands.ci.cpp]: Found a couple of other spots that isn't supported in VS2015.

File: toolsrc/include/vcpkg/dependencies.h [hunk 3]
```diff
@@ -47,12 +47,24 @@ namespace vcpkg::Dependencies
                           const std::set<std::string>& features,
                           const RequestType& request_type);
 
+        InstallPlanAction(InstalledPackageView&& spghs,
+                          const std::set<std::string>& features,
+                          const RequestType& request_type,
+                          const Build::BuildPackageOptions&);
```
Comment (Rastaban) [inline toolsrc/include/vcpkg/dependencies.h]: give this a name so it matches with the other parameters.

File: toolsrc/include/vcpkg/dependencies.h [hunk 4]
```diff
@@ -181,18 +194,43 @@ namespace vcpkg::Dependencies
...
+                                                       const StatusParagraphs& status_db)
+    {
+        MapPortFileProvider provider(map);
+        return create_feature_install_plan(provider, specs, status_db);
+    }
 
     /// <summary>Figure out which actions are required to install features specifications in `specs`.</summary>
     /// <param name="provider">Contains the ports of the current environment.</param>
     /// <param name="specs">Feature specifications to resolve dependencies for.</param>
     /// <param name="status_db">Status of installed packages in the current environment.</param>
```
Comment (Rastaban) [inline toolsrc/include/vcpkg/dependencies.h]: out of date (although it looks like it was also out of date before your changes too).

Comment (cbezault) [reply toolsrc/include/vcpkg/dependencies.h]: Are you referring to the comment?

**Comment by cbezault** [comment]
reopening to test something in the ci
